### ISSUE TITLE
[BUG][JAVA] Enum generation fails on same values in upperCase and lowerCase

### ISSUE THREAD

**Issue by mbreevoort** [OpenAPITools/openapi-generator#12624]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description
For Java the enum's are generated with name upper and value is the value of the openapi spec.
In our case we need to be compliant with another specification which has the same value in uppercase and lowercase.
This causes a compile error in the java generated enum:
```
MyObject.java:90: error: variable S is already defined in enum UnitEnum
    S("S"),
    ^
1 error
```

Is there a config setting for this? (Could not find it)

Actual:
Compile exception (see above)

Expected:
Generation of multiple casing should compile the java enum file
Maybe just like `org.jsonschema2pojo.Jsonschema2Pojo` It generates a enum name with an underscore `S_("S"),` for the second name.

##### openapi-generator version

'org.openapi.generator' version '5.4.0'

##### OpenAPI declaration file content or url

```yaml
    MyObject:
      type: object
      properties:
        description:
          type: string
        unit:
          type: string
          enum:
          - s
          - S
```

##### Generation Details

gradle plugin is used

##### Steps to reproduce

Generate the models

##### Related issues/PRs

In this issue a same problem is described but not solved: https://github.com/OpenAPITools/openapi-generator/issues/12187#issuecomment-1104862402

##### Suggest a fix

Maybe just like `org.jsonschema2pojo.Jsonschema2Pojo` It generates a enum name with an underscore `S_("S"),` for the second name.

**Comment by dmmarques** [comment]
I am currently facing the same issue, have you had any updates on this ?

**Comment by mbreevoort** [comment]
No updates.
I don't know if it helps to create a pull request for this when no maintainer responds...

**Comment by wing328** [comment]
Just responded to a similar PR/issue but for C#: https://github.com/OpenAPITools/openapi-generator/pull/16981#issuecomment-1793422922

If we add the `enumNameMapping` option, it should fix the issue, right?

**Comment by dmmarques** [comment]
Does it add a suffix when both generated values are the same ?

**Comment by wing328** [comment]
It doesn't. Using the example provide in the issue. the command line input would look like

```
--enumNameMapping s=LOWER_CASE_S,S=UPPER_CASE_S
```

so that users will have complete control of the enum naming to handle edge cases reported in this issue.

**Comment by wing328** [comment]
Merged #17018 to fix the issue

Please pull the latest master to give it a try.
